### ISSUE TITLE
Bug with param matching in beta.7

### ISSUE THREAD

**Issue by zackify** [ReactTraining/react-router#4664]
## Version
v4 beta 7

## Test Case
I was going to attempt to make one, but this jsbin was for an old version.

Here's a copy paste example that you can put into create-react-app:

```js
import React from 'react';
import { BrowserRouter as Router, Route, Link, withRouter } from 'react-router-dom';

const BasicExample = () => (
  <Router>
    <div>
      <ul>
        <li><Link to="/">Home</Link></li>
        <li><Link to="/about">About</Link></li>
        <li><Link to="/topics">Topics</Link></li>
      </ul>

      <hr />

      <Route exact path="/" component={Home} />
      <Route path="/about" component={About} />
      <Route path="/topics/:name?" component={Topics} />
    </div>
  </Router>
);

const Home = () => (
  <div>
    <h2>Home</h2>
  </div>
);

const About = () => (
  <div>
    <h2>About</h2>
  </div>
);

const Topics = ({ match }) => {
  console.log(match);
  return (
    <div>
      <h2>Topics</h2>
      <ul>
        <li><Link to={`${match.url}/rendering`}>Rendering with React</Link></li>
        <li><Link to={`${match.url}/components`}>Components</Link></li>
        <li><Link to={`${match.url}/props-v-state`}>Props v. State</Link></li>
      </ul>
      <TestComponent />
      <Route path={`${match.url}/:topicId`} component={Topic} />
      <Route exact path={match.url} render={() => <h3>Please select a topic.</h3>} />
    </div>
  );
};

const Topic = ({ match }) => (
  <div>
    <h3>{match.params.topicId}</h3>
  </div>
);

const TestComponent = withRouter(props => {
  console.log(props);
  return <div>test</div>;
});

export default BasicExample;

```

Go to `/topics/blah` and view the console to see that the params are not in the nested component.

## Steps to reproduce

Log the `match` object at the top level component rendered inside of a route. Then, in a component nested inside of the route, use `withRouter` and the match will not contain any params:

<img width="337" alt="screen shot 2017-03-08 at 11 45 54 am" src="https://cloud.githubusercontent.com/assets/449136/23713735/d218c780-03f4-11e7-81af-aee7510c8552.png">

## Expected Behavior

for match to work in `withRouter`

## Actual Behavior

it doesn't :)

**Comment by pshrmn** [comment]
Already fixed in 394eb09 / f90d753

**Comment by zackify** [comment]
Any idea on when a release will go out, that's why I didn't realize it was fixed. @pshrmn
